interop/xds/client.go

```diff
@@ -0,0 +1,181 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net"
+	"strconv"
+	"sync"
+	"time"
+
+	"google.golang.org/grpc"
+	_ "google.golang.org/grpc/balancer/grpclb"
+	"google.golang.org/grpc/grpclog"
+	testpb "google.golang.org/grpc/interop/grpc_testing"
+	"google.golang.org/grpc/peer"
+)
+
+type statsWatcherKey struct {
+	startId int32
+	endId   int32
+}
+
+type statsWatcher struct {
+	rpcsByPeer    map[string]int32
+	numFailures   int32
+	remainingRpcs int32
+	lock          sync.Mutex
+}
+
+var (
+	dialTimeout   = flag.Duration("dialTimeout", 10*time.Second, "timeout for creating grpc.ClientConn")
+	numChannels   = flag.Int("num_channels", 1, "Num of channels")
+	printResponse = flag.Bool("print_response", false, "Write RPC response to stdout")
+	qps           = flag.Int("qps", 1, "QPS (across all channels)")
+	rpcTimeout    = flag.Duration("rpc_timeout", 2*time.Second, "Per RPC timeout")
+	server        = flag.String("server", "localhost:8080", "Address of server to connect to")
+	statsPort     = flag.Int("stats_port", 8081, "Port to expose peer distribution stats service")
+
+	currentRequestId int32 = 0
+	lock             sync.Mutex
+	watchers         = make(map[statsWatcherKey]*statsWatcher)
+)
+
+type statsService struct{}
+
+func (s *statsService) GetClientStats(ctx context.Context, in *testpb.LoadBalancerStatsRequest) (*testpb.LoadBalancerStatsResponse, error) {
```

Comment:
The behavior of this method isn't very clear to me. Can you add some comments or link to the spec if there's one?